packages/web/components/bridge/immersive/amount-screen.tsx

```diff
@@ -0,0 +1,472 @@
+import { Menu } from "@headlessui/react";
+import { CoinPretty, Dec, DecUtils, PricePretty } from "@keplr-wallet/unit";
+import { DEFAULT_VS_CURRENCY } from "@osmosis-labs/server";
+import { MinimalAsset } from "@osmosis-labs/types";
+import { isNumeric, noop } from "@osmosis-labs/utils";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import Image from "next/image";
+import { FunctionComponent, ReactNode, useState } from "react";
+
+import { Icon } from "~/components/assets";
+import { BridgeNetworkSelect } from "~/components/bridge/immersive/bridge-network-select";
+import { MoreBridgeOptions } from "~/components/bridge/immersive/more-bridge-options";
+import { InputBox } from "~/components/input";
+import { SkeletonLoader, Spinner } from "~/components/loaders";
+import { Tooltip } from "~/components/tooltip";
+import { Button } from "~/components/ui/button";
+import { useConnectWalletModalRedirect, useTranslation } from "~/hooks";
+import { usePrice } from "~/hooks/queries/assets/use-price";
+import { useStore } from "~/stores";
+import { trimPlaceholderZeros } from "~/utils/number";
+import { api } from "~/utils/trpc";
+
+interface AmountScreenProps {
+  type: "deposit" | "withdraw";
+
+  /**
+   * Includes both the canonical asset and its variants.
+   */
+  assetsInOsmosis: MinimalAsset[] | undefined;
```

Comment:
Nit: what if we call this `assetWithVariants`?

Reply:
Hmm, my initial thought with this name is that it suggests an asset object with nested variants. Since it's an array, perhaps it could be called `canonicalAndVariantAssets`